### PR TITLE
Register set operations

### PR DESCRIPTION
**Pull request by dpsanders**

### CODE REVIEW

File: src/extra_functions.jl [hunk 1]
```diff
@@ -11,3 +11,12 @@ ModelingToolkit.derivative(::typeof(abs), args::NTuple{1,Any}, ::Val{1}) = IfEls
...
 ModelingToolkit.derivative(::typeof(IfElse.ifelse), args::NTuple{3,Any}, ::Val{2}) = IfElse.ifelse(args[1],1,0)
 ModelingToolkit.derivative(::typeof(IfElse.ifelse), args::NTuple{3,Any}, ::Val{3}) = IfElse.ifelse(args[1],0,1)
+
+@register ∈(x::Num, y::AbstractArray)
+
+@register ∪(x, y)
+@register ∩(x, y)
+@register ∨(x, y)
+@register ∧(x, y)
+@register ≠(x, y)
```
Comment (YingboMa) [inline src/extra_functions.jl:21]: This is not right. Don't we already have `x != y`?

File: src/extra_functions.jl [hunk 2]
```diff
@@ -11,3 +11,11 @@ ModelingToolkit.derivative(::typeof(abs), args::NTuple{1,Any}, ::Val{1}) = IfEls
 ModelingToolkit.derivative(::typeof(IfElse.ifelse), args::NTuple{3,Any}, ::Val{1}) = 0
 ModelingToolkit.derivative(::typeof(IfElse.ifelse), args::NTuple{3,Any}, ::Val{2}) = IfElse.ifelse(args[1],1,0)
 ModelingToolkit.derivative(::typeof(IfElse.ifelse), args::NTuple{3,Any}, ::Val{3}) = IfElse.ifelse(args[1],0,1)
+
+@register ∈(x::Num, y::AbstractArray)
+
+@register ∪(x, y)
```
Comment (YingboMa) [inline src/extra_functions.jl:17]: I am not sure if this is correct. We have `Num <: Real`, so `∪(x, y)` doesn't really make sense.

Comment (dpsanders) [reply src/extra_functions.jl:43]: It makes sense for intervals...

Comment (dpsanders) [reply src/extra_functions.jl:43]: Although it's not clear that it makes sense for intervals to be `<: Real`.

Comment (ChrisRackauckas) [reply src/extra_functions.jl:43]: I'll take it. It won't be useful to trace through, that's for sure.

**Pull request merged**